logsmanagement/db_api.c

```diff
@@ -0,0 +1,1348 @@
+            collector_info("[%s]: About to execute " METADATA_DB_FILENAME " migrations", p_file_info->chart_name);
+            int metadata_db_ver = db_user_version(p_file_info->db, -1);
+            if (likely(LOGS_MANAG_DB_VERSION == metadata_db_ver)) {
+                collector_info( "[%s]: Logs management " METADATA_DB_FILENAME " database version is %d (no migration needed)", 
+                                p_file_info->chart_name, metadata_db_ver);
+            } else {
+                for(int ver = metadata_db_ver; ver < LOGS_MANAG_DB_VERSION && migration_list_metadata_db[ver].func; ver++){
+                    rc = (migration_list_metadata_db[ver].func)(p_file_info->db, migration_list_metadata_db[ver].name);
+                    if (unlikely(rc)){
+                        collector_error("[%s]: Logs management " METADATA_DB_FILENAME " database migration from version %d to version %d failed", 
+                                        p_file_info->chart_name, ver, ver + 1);
+                        throw_error(MAIN_DB, ERR_TYPE_SQLITE, rc, __LINE__, __FILE__, __FUNCTION__);
+                        uv_mutex_unlock(p_file_info->db_mut);
+                        goto return_error;
+                    }
+                    db_user_version(p_file_info->db, ver + 1);
+                }
+            }
+
+            /* -----------------------------------------------------------------
+             * Create BLOBS_TABLE and LOGS_TABLE if they don't exist. Do it 
+             * as a transaction, so that it can all be rolled back if something
+             * goes wrong.
+             * -------------------------------------------------------------- */
+            {
+                rc = sqlite3_exec(p_file_info->db, "BEGIN TRANSACTION;", NULL, NULL, NULL);
+                do_sqlite_error_check(p_file_info, rc, SQLITE_OK);
+
+                /* Check if BLOBS_TABLE exists or not */
+                sqlite3_stmt *stmt_check_if_BLOBS_TABLE_exists;
+                rc = sqlite3_prepare_v2(p_file_info->db,
+                                        "SELECT COUNT(*) FROM sqlite_master" 
+                                        " WHERE type='table' AND name='"BLOBS_TABLE"';",
+                                        -1, &stmt_check_if_BLOBS_TABLE_exists, NULL);
+                do_sqlite_error_check(p_file_info, rc, SQLITE_OK);
+                rc = sqlite3_step(stmt_check_if_BLOBS_TABLE_exists);
+                do_sqlite_error_check(p_file_info, rc, SQLITE_ROW);
+                
+                /* If BLOBS_TABLE doesn't exist, create and populate it */
+                if(sqlite3_column_int(stmt_check_if_BLOBS_TABLE_exists, 0) == 0){
+                    
+                    /* 1. Create it */
+                    rc = sqlite3_exec(p_file_info->db,
+                            "CREATE TABLE IF NOT EXISTS " BLOBS_TABLE "("
+                            "Id         INTEGER     PRIMARY KEY,"
```

Comment:
An integer as primary key means that your code needs to control what db can control, why are you not using `uuid` or `autoincrement`? Both suggestions are not perfect, but probably they reduce concern on your end.
Please, create an index for `Filename`.

Reply:
I don't believe that's necessary in this case, [SQLite docs](https://www.sqlite.org/autoinc.html) actually advise against using `AUTOINCREMENT` unless strictly necessary. `INTEGER PRIMARY KEY` should suffice as it is an alias for `ROWID`. 

 About `FILENAME`, we can have a max of 10 BLOBs at the moment which get rotated and even in the future, it is not expected to have _too_ many files to justify the use of an index. I mean, I can add one, but it just adds to code complexity without significant gains.